tools/install_dev_dependencies.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# Installs dependencies required for LORIS development on Ubuntu
+
+os_distro=$(hostnamectl |awk -F: '/Operating System:/{print $2}'|cut -f2 -d ' ')
+debian=("Debian" "Ubuntu")
+if [[ ! " ${debian[*]} " =~ " $os_distro " ]]; then
+    echo "Only Debian and Ubuntu are supported by $0."
+    exit 0;
+fi
+
+# apache2 is not included here because a developer may use a different server
+# config, such as the builtin PHP web server.
+declare -a prereqs=(
+"php"
```

Comment:
The script should be able to install php. It should be installed by dependecies for some of the packages in the second part (php$PHP-...).

It should not be a case of failure.

Reply:
I think that's a bit tricky because the version of PHP used by the command line isn't necessarily the same as the one Apache2 is using. I think this part should be left to the person setting up the system, similar to how it is done now.

Reply:
Yes, and in the second part, even if you install php7.2-xyz package, Apache might use php7.0-xyz package. 
To know which version Apache is using, you need to get to call phpinfo() (from the web server) and parse the result.
Maybe we could add a page with this function accessible only from localhost. That would required more apache config which could cause problem in multi-site hosting.

Reply:
The point of the PR is just to install system dependencies that so far don't get automatically installed. It's not intended to configure apache but to check that the right packages are installed.

Reply:
So why fail when PHP is not installed?

Reply:
These are prerequisites for LORIS that we are not configuring for people, but just documenting:

> Apache 2.4 or higher
MySQL >= 5.7 (or MariaDB >= 10.3)
PHP 7.2 or higher
Composer
NodeJS 8.0 or higher

We also require these libraries, but they are not well-documented:

> "curl"
"wget"
"zip"
"unzip"
"php-json"
"make"
"software-properties-common"
"php-ast"
"php$PHP"
"php$PHP-mysql"
"php$PHP-xml"
"php$PHP-json"
"php$PHP-mbstring"
"php$PHP-gd"
"libapache2-mod-php$PHP"

I'm looking to install the second list. They are easy to install and can be done by a simple bash script.

If we get into trying to automatically configure the entire LAMP stack for developers than we are *much* better off working toward some kind of containerization.

I'm in favour of making the LORIS install as easy as possible but frankly this is my third complete rewrite of trying to make the developer install easier. It seems that there isn't the time or will within the team to review and test a more complex/robust developer set-up. That's fine, that's the reality of it, but the problem remains that many people are tripping over these developer dependencies.

I'm keeping this script simple because I want it to have a chance of getting merged and fixing this problem that I've been trying to remedy for a year and a half. 

If you'd like to work on something more robust please go ahead and I'll be the first to review it.